From: Sungjong Seo <sj1557.seo@samsung.com>
Date: Wed, 26 Mar 2025 23:48:48 +0900
Subject: [PATCH] exfat: fix potential wrong error return from get_block
References: bsc#1234429
Patch-mainline: 6.12.23
Git-commit: 59c30e31425833385e6644ad33151420e37eabe1

commit 59c30e31425833385e6644ad33151420e37eabe1 upstream.

If there is no error, get_block() should return 0. However, when bh_read()
returns 1, get_block() also returns 1 in the same manner.

Let's set err to 0, if there is no error from bh_read()

Fixes: 11a347fb6cef ("exfat: change to get file size from DataLength")
Cc: stable@vger.kernel.org
Signed-off-by: Sungjong Seo <sj1557.seo@samsung.com>
Reviewed-by: Yuezhang Mo <Yuezhang.Mo@sony.com>
Signed-off-by: Namjae Jeon <linkinjeon@kernel.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: Robert Frohl <rfrohl@suse.com>
---
 fs/exfat/inode.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/fs/exfat/inode.c b/fs/exfat/inode.c
index 6e13c597e984..3801516ac507 100644
--- a/fs/exfat/inode.c
+++ b/fs/exfat/inode.c
@@ -391,6 +391,8 @@ static int exfat_get_block(struct inode *inode, sector_t iblock,
 			/* Zero unwritten part of a block */
 			memset(bh_result->b_data + size, 0,
 			       bh_result->b_size - size);
+
+			err = 0;
 		} else {
 			/*
 			 * The range has not been written, clear the mapped flag
-- 
2.49.0

